src/contracts/SavingCircles.sol

```diff
@@ -223,11 +245,9 @@ contract SavingCircles is ISavingCircles, ReentrancyGuard, OwnableUpgradeable {
    *      A deposit must be made in specific time window and can be made partially so long as the final balance equals
    *      the specified deposit amount for the circle.
    */
-  function _deposit(uint256 _id, uint256 _value, address _member) internal {
+  function _deposit(uint256 _id, uint256 _value, address _member) internal onlyCommissioned(_id) onlyMember(_id) {
     Circle memory _circle = circles[_id];
 
-    if (_isDecommissioned(_circle)) revert NotCommissioned();
-    if (!isMember[_id][_member]) revert NotMember();
```

Comment:
`onlyMember(_id)` modifier is not a valid replacement for this line. The modifier assumes the address to check is the caller, but that isn't the case here. You can, update the modifier to be `onlyMember(uint256 _id, address _member)` instead